meilisearch-http/tests/content_type.rs

```diff
@@ -0,0 +1,109 @@
+mod common;
+
+use crate::common::{GetAllDocumentsOptions, Server};
+use actix_web::test;
+use meilisearch_http::create_app;
+use serde_json::{json, Value};
+
+#[actix_rt::test]
+async fn strict_json_bad_content_type() {
+    let routes = [
+        // all the POST route except the dumps that can be created without any body or content-type
+        // and the search that is not a strict json
+        "/indexes",
+        "/indexes/doggo/documents/delete-batch",
+        "/indexes/doggo/search",
+        "/indexes/doggo/settings",
+        "/indexes/doggo/settings/displayed-attributes",
+        "/indexes/doggo/settings/distinct-attribute",
+        "/indexes/doggo/settings/filterable-attributes",
+        "/indexes/doggo/settings/ranking-rules",
+        "/indexes/doggo/settings/searchable-attributes",
+        "/indexes/doggo/settings/sortable-attributes",
+        "/indexes/doggo/settings/stop-words",
+        "/indexes/doggo/settings/synonyms",
+    ];
+    let bad_content_types = [
+        "application/csv",
+        "application/x-ndjson",
+        "application/x-www-form-urlencoded",
+        "text/plain",
+        "json",
+        "application",
+        "json/application",
+    ];
+
+    let document = "{}";
+    let server = Server::new().await;
+    let app = test::init_service(create_app!(
+        &server.service.meilisearch,
+        true,
+        &server.service.options
+    ))
+    .await;
+    for route in routes {
```

Comment:
bonus: the partition can be made even more explicit with https://docs.rs/itertools/0.10.1/itertools/trait.Itertools.html#method.cartesian_product :) (i know you like iterators)

Reply:
I could, but the problem is that I make two tests independently of the content-type (the no content-type and the good content-type tests).

If I understood correctly, the cartesian_product would not help me here, or I would need to redo these two tests for every content_type instead of doing it only once?